pkg/netns/netns_linux.go

```diff
@@ -138,6 +138,13 @@ func NewNS() (ns.NetNS, error) {
 		if err != nil {
 			err = fmt.Errorf("failed to bind mount ns at %s: %v", nsPath, err)
 		}
+
+		// mount sysfs so the process running in the container can see the
+		// correct /sys/class/net entries
+		err = unix.Mount("sysfs", "/sys", "sysfs", unix.MS_RDONLY, "")
```

Comment:
The goroutine is running in a new network namespace but it is sharing the mount namespace with the host.  How is the new mount visible from the container?

Reply:
I may be very off in my understanding of namespaces, but by default doesn't podman create a user namespace that all containers share (unless explicitly stated). The network namespace is being created inside that user namespace, and since mount namespaces require an owner user namespace all that information would be accessible by the container if it enters the network namespace?

Reply:
blah disregard that. I think what's happening is that it is being mounted correctly, but it needs to do a mount sooner, right after the network namespace creation in order to the `/sys/class/net` to be generated correctly for the oci runtime to use.

Reply:
@gabibeyer Perhaps, the commit message needs to be rephrased a bit, to specify that the OCI runtime needs to access the /sys.
@giuseppe We need this behaviour for the runtime. In case of Kata, we need to inspect the network namespace to get the type of network interfaces created in order to connect them to the virtual machine.
Most of this information can be obtained from netlink API.
But we may need to get this from sysfs in some cases. In case of tap interfaces, we do not get the information whether a tuntap device is actually "tap" or "tun" on older kernels from netlink. In that case we need to fallback to the sysfs, and expect the sysfs to be remounted to reflect the correct `/sys/class/net` information corresponding to the new network namespace created by podman.

I hope that makes things a bit clearer. A related PR submitted to netlink:
https://github.com/vishvananda/netlink/commit/db99c040b9b1548675e396ed28c1bbd47573efa5